train/training.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import os
+import csv
+import requests
+from azure.storage.blob import BlockBlobService, ContentSettings
+from utils.config import Config
+from utils.blob_utils import BlobStorage
+import urllib.request
+import sys
+import time
+
+CONFIG_PATH = os.environ.get('ALCONFIG', None)
+
+def train(config, num_images):
+
+    # First, downloxad data necessary for training
+    training_data = download_data_for_training(config, num_images)
+
+    # Make sure directory is clean:
+    file_location = Config.initialize_tagging_location(config)
+
+    # Grab tagged and totag images from the blob storage
+    download_images(training_data["imageURLs"], str(file_location))
+
+    # create csv file from this data
+    convert_to_csv(training_data, str(file_location))
+
+
+def download_images(imageURLs, file_location):
+    print("Downloading images to " + file_location + ", this may take a few seconds...")
+
+    # Download tagged images into tagged folder
+    if not os.path.exists(file_location + '/AllImages'):
+        os.makedirs(file_location + '/AllImages')
+    folder = file_location + '/AllImages'
+    for image in imageURLs:
+        filename = get_image_name_from_url(image)
+        location = image
+        # extension = location.split('.')[-1]
+        with urllib.request.urlopen(location) as response, open(folder + '/' + str(filename), 'wb') as out_file:
+            data = response.read() # a `bytes` object
+            out_file.write(data)
+    
+    print("Downloaded images into " + file_location + "/AllImages/")
+
+
+def download_data_for_training(config, num_images):
+    print("Downloading data for training, this may take a few moments...")
+    # Download n images that are ready to tag
+    query = {
+        "userName": config.get('tagging_user'),
+        "imageCount": num_images,
+        "tagStatus": 1
+    }
+    url = config.get('url') + '/api/images'
+    response = requests.get(url, params=query)
+    to_tag_image_info = response.json()
+    image_urls_to_download = [info['location'] for info in to_tag_image_info]
+
+    # Download upto 200 images that have been tagged, for training
+    query['tagStatus'] = 3
+    query['imageCount'] = 200
+    url = config.get('url') + '/api/tags'
```

Comment:
@samiyaakhtar I mentioned this in another comment but it might make more sense to be driven by the labels (what you're calling tags) then download the images based upon the image ids in the tags. The current way doesn't seem to guarantee that the image and label data will be correlated.